eta/detectors/tfmodels_segmenters.py

```diff
@@ -0,0 +1,197 @@
+'''
+Interface to the instance segmentation models from the TF object detection
+library available at
+https://github.com/tensorflow/models/tree/master/research/object_detection.
+
+Copyright 2017-2019, Voxel51, Inc.
+voxel51.com
+
+Brian Moore, brian@voxel51.com
+'''
+# pragma pylint: disable=redefined-builtin
+# pragma pylint: disable=unused-wildcard-import
+# pragma pylint: disable=wildcard-import
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from builtins import *
+# pragma pylint: enable=redefined-builtin
+# pragma pylint: enable=unused-wildcard-import
+# pragma pylint: enable=wildcard-import
+
+import os
+import sys
+
+import numpy as np
+import tensorflow as tf
+
+import eta.constants as etac
+from eta.core.config import Config, ConfigError
+from eta.core.geometry import BoundingBox, RelativePoint
+from eta.core.learning import ObjectDetector, HasDefaultDeploymentConfig
+import eta.core.models as etam
+from eta.core.objects import DetectedObject, DetectedObjectContainer
+from eta.core.tfutils import UsesTFSession
+import eta.core.utils as etau
+
+sys.path.append(os.path.join(etac.TF_OBJECT_DETECTION_DIR, "utils"))
+import label_map_util as gool
+
+
+class TFModelsSegmenterConfig(Config, HasDefaultDeploymentConfig):
+    '''TFModelsSegmenter configuration settings.
+
+    Note that `labels_path` is passed through
+    `eta.core.utils.fill_config_patterns` at load time, so it can contain
+    patterns to be resolved.
+
+    Note that this class implements the `HasDefaultDeploymentConfig` mixin, so
+    if a published model is provided via the `model_name` attribute, then any
+    omitted fields present in the default deployment config for the published
+    model will be automatically populated.
+
+    Attributes:
+        model_name: the name of the published model to load. If this value is
+            provided, `model_path` does not need to be
+        model_path: the path to a TF SavedModel in `.pb` format to load. If
```

Comment:
To clarify, this would be `my_downloaded_model/saved_model/saved_model.pb`, and not the `my_downloaded_model/frozen_inference_graph.pb` ?

Reply:
It is the frozen inference graph, not the saved model. Good catch. Updated.